src/Domain/Domain.cpp

```diff
@@ -27,13 +27,28 @@ template <size_t VolumeDim>
 Domain<VolumeDim>::Domain(
     std::vector<std::unique_ptr<
         domain::CoordinateMapBase<Frame::Logical, Frame::Inertial, VolumeDim>>>
-        maps) noexcept {
+        maps,
+    std::vector<DirectionMap<
+        VolumeDim,
+        std::unique_ptr<domain::BoundaryConditions::BoundaryCondition>>>
+        boundary_conditions) noexcept {
+  ASSERT(
+      boundary_conditions.size() == maps.size() or boundary_conditions.empty(),
+      "There must be either one boundary condition per block or none at all "
```

Comment:
`...one set of boundary conditions per...` or similar (another occurrence below)